### ISSUE TITLE
Filter matching context by last modification date

### ISSUE THREAD

**Issue by lutangar** [dis-moi/backend#431]
> @felix-lambert said: Would that be complicated to only fetch modified matching context?

On Api V4: https://api.dismoi.io/v4 it would be rather easy to add such filters, see api-platform documentation:
https://api-platform.com/docs/core/filters/#search-filter

But first we must add 2 properties on the `MatchingContext` entity. Take example on the `Notice` entity:
https://github.com/dis-moi/backend/blob/master/src/Entity/Notice.php#L257-L282

Then some logic must be aded to update both dates, again you may take example on the `NoticeListener`, see the listener and the required annotation :
https://github.com/dis-moi/backend/blob/master/src/Entity/Notice.php#L33

**Comment by JalilArfaoui** [comment]
What about deletions ?

**Comment by lutangar** [comment]
Could be done via a *soft delete* feature with a `deletedAt` properties and why not a `isDeleted` getter.
I think you mentioned another solution that might be better to handle the usecase... feel free to edit the title of the feature to better suit the usecase description, this title is too precise and emphasize a possible solution and not the actual issue.

**Comment by JalilArfaoui** [comment]
@lutangar Yes, a soft delete would enable us to set up a differential matching contexts update …

I’m not sure that this is high-priority though…
